junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertionsAssertArrayEqualsTests.java

```diff
@@ -1874,8 +1876,8 @@ void assertArrayEqualsDifferentNestedObjectArrays() {
 			expectAssertionFailedError();
 		}
 		catch (AssertionFailedError ex) {
-			assertMessageEquals(ex,
-				"array contents differ at index [3][0][1][0], expected: <5> but was: <" + differentElement + ">");
+			assertMessageEquals(ex, "array contents differ at index [3][0][1][0], expected: <5> but was: <"
+					+ Arrays.toString(differentElement) + ">");
```

Comment:
Oh yes, good catch!

I've fixed this locally, intending to push it as a fixup commit soon. :)